client/src/components/QuestionPopup.tsx

```diff
@@ -0,0 +1,217 @@
+import { correctAnswer, wrongAnswer } from "~/questions/QuestionLogic";
+
+export class QuestionPopup {
+  scene: any;
+  popup: any;
+  input: any;
+  confirmButton: any;
+  onSubmit: any;
+  textLabel: any;
+  scrollablePanel: any;
+  options: string[];
+  optionBoxes: any[]; // Array to keep references to option graphics and text
+  closeButton: any; // Reference to the close button
+  container: Phaser.GameObjects.Container;
+
+  constructor(scene) {
+    this.scene = scene;
+    this.popup = null;
+    this.input = null;
+    this.confirmButton = null;
+    this.textLabel = null;
+    this.scrollablePanel = null;
+    this.options = [];
+    this.optionBoxes = []; // Initialize the array
+    this.closeButton = null;
+
+    // Create the container and position it in the center of the camera's viewport
+  }
+  createPopup() {
+    const popupOffset = { x: 0, y: -50 }; // Adjust as needed
+    const screenCenterX = this.scene.cameras.main.centerX;
+    const screenCenterY = this.scene.cameras.main.centerY;
+
+    this.container = this.scene.add.container();
+    console.log();
+    // this.container.setScrollFactor(0);
+    const popupWidth = 600; // Adjusted for larger content
+    const popupHeight = 500;
+    const x = this.scene.cameras.main.centerX;
+    // const y =
+    //   this.scene.cameras.main.worldView.y +
+    //   this.scene.cameras.main.height / 1.5 -
+    //   80;
+    const y = this.scene.cameras.main.centerY;
+    this.container.setScrollFactor(0);
+
+    // Popup Background
+    this.popup = this.scene.add
+      .graphics({
+        x: x - popupWidth / 2,
+        y: y - popupHeight / 2,
+      })
+      .fillStyle(0x000000, 0.8)
+      .fillRoundedRect(0, 0, popupWidth, popupHeight, 20);
+
+    const closeButton = this.scene.add
+      .text(x + popupWidth / 2 - 20, y - popupHeight / 2 + 5, "X", {
+        fontSize: "20px",
+        color: "#ffffff",
+        backgroundColor: "#ff0000",
+        padding: {
+          left: 5,
+          right: 5,
+          top: 5,
+          bottom: 5,
+        },
+      })
+      .setInteractive();
+
+    // Close button functionality
+    closeButton.on("pointerdown", () => {
```

Comment:
very small nit but can we do pointerDown instead

Reply:
i think this is fixed